readme.md

```diff
@@ -691,7 +696,75 @@ coordsToPair(pairToCoords([1, 2])) // [1, 2]
 pairToCoords(coordsToPair({x: 1, y: 2})) // {x: 1, y: 2}
 ```
 
+### Catamorphism
 
+A `reduceRight` function that applies a function against an accumulator and each value of the array (from right-to-left) to reduce it to a single value.
```

Comment:
Back to your original question. I'm not sure how "generalised" definition we should provide here either... Looking at other places they all seem to be simplified definitions (thus not strictly correct). I'm not sure how we balance correctness and ease to understand. @hemanth thoughts?